### ISSUE TITLE
CSSStyleSheet instanceof check does not work

### ISSUE THREAD

**Issue by Lodin** [calebdwilliams/construct-style-sheets#4]
I tried to create a simple `instanceof` check for the `CSSStyleSheet` object, but it returns `false`.
Is it possible to pass the check? 
```javascript
const foo = new CSSStyleSheet();
console.log(foo instanceof CSSStyleSheet); // => false
```

**Comment by calebdwilliams** [comment]
Yeah, this one is a bit weird. Both Firefox and Safari have the actual style sheet object (ie `document.styleSheets[0]` as instances of the `StyleSheet` global rather than `CSSStyleSheet`. I don't really know why that is.

Supposedly you can tinker with the prototype to make this call work. I wasn't able to in the five minutes I tried, but even if I could I question the wisdom in doing so as that could potentially cause issues with things like the objects methods (`addRule`, etc.).

I'll leave this issue open and keep playing with it, but I'm not going to have a ton of time this week to look at this. If you are interested in taking a crack at it, feel free and let me know what you learn.

**Comment by Lodin** [comment]
Oh I missed your reply. I found the case and the solution. Basically, it was because you replaces `window.CSSStyleSheet` with a `_StyleSheet` class, and then return an actual ("old") `CSSStyleSheet` from its constructor. So while the `instanceof` check I expect the instance of the `CSSStyleSheet` to be an instance of the `_StyleSheet` class under the `window.CSSStyleSheet`. That's why it fails. 

I solved it by simple `instanceof` overriding via the `Symbol.hasInstance`. Works as expected.
